bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -111,15 +139,33 @@ private SecurityContext authenticateBearerToken(String token) throws Authenticat
         }
     }
 
-    private SecurityContext authenticateUsernamePassword(String username, String password)
+    private SecurityContext authenticateUsernamePassword(String username, String password, String authCacheKey)
```

Comment:
`authCacheKey` isn't a cache key, it is the unencrypted credentials. The parameter name is hence dangerously misleading.

Reply:
Besides this, I do not see any need for this third parameter, since it holds no additional information over the first two.

Reply:
But why split the String and then join it back together - this seems like unnecessary overhead?
If I pass it into the function I can just hash it the way it is.

Reply:
What would be a better name for the parameter? It gets hashed and used as the key ...

Reply:
Maybe the method should only receive the BASE64 credential string as a param - that would be simplest.